File: native/Travel Companion/Model/Flickr/FlickrConstants.swift
//
//  ParseConstants.swift
//  VirtualTourist
//
//  Created by Stefan Jaindl on 23.05.18.
//  Copyright © 2018 Stefan Jaindl. All rights reserved.
//

import Foundation

struct FlickrConstants {

    struct UrlComponents {
        static let urlProtocol = "https"
        static let domain = "api.flickr.com"
        static let path = "/services/rest"
    }
    
    struct Location {
        static let searchBBoxHalfWidth = 0.01
        static let searchBBoxHalfHeight = 0.01
        static let searchLatitudeRange = (-90.0, 90.0)
        static let searchLongitudeRange = (-180.0, 180.0)
    }
    
    struct ParameterKeys {
        static let method = "method"
        static let apiKey = "api_key"
        static let extras = "extras"
        static let format = "format"
        static let noJsonCallback = "nojsoncallback"
        static let sortOrder = "sort"
        static let safeSearch = "safe_search"
        static let text = "text"
        static let boundingBox = "bbox"
    }
    
    struct ParameterValues {
        static let searchMethod = "flickr.photos.search"
        static let responseFormat = "json"
        static let disableJsonCallback = "1" /* 1 means "yes" */
        static let imageSize = "url_m"
        static let useSafeSearch = "1"
        static let sortOrder = "relevance"
    }
    
    struct ResponseKeys {
        static let photos = "photos"
        static let photo = "photo"
        static let title = "title"
        static let imageSize = "url_m"
    }
}

File: shared/src/commonMain/kotlin/com.sjaindl.travelcompanion/api/flickr/FlickrConstants.kt
package com.sjaindl.travelcompanion.api.flickr

object FlickrConstants {
    object UrlComponents {
        const val urlProtocol = "https"
        const val domain = "api.flickr.com"
        const val path = "/services/rest"
    }

    object Location {
        const val searchBBoxHalfWidth = 0.01
        const val searchBBoxHalfHeight = 0.01
        val searchLatitudeRange = -90.0 to 90.0
        val searchLongitudeRange = -180.0 to 180.0
    }

    object ParameterKeys {
        const val method = "method"
        const val apiKey = "api_key"
        const val extras = "extras"
        const val format = "format"
        const val noJsonCallback = "nojsoncallback"
        const val sortOrder = "sort"
        const val safeSearch = "safe_search"
        const val text = "text"
        const val boundingBox = "bbox"

        const val limit = "per_page"
        const val offset = "page"
    }

    object ParameterValues {
        const val searchMethod = "flickr.photos.search"
        const val responseFormat = "json"
        const val disableJsonCallback = "1" /* 1 means "yes" */
        const val imageSize = "url_m"
        const val useSafeSearch = "1"
        const val sortOrder = "relevance"

        const val limit = 20
    }

    object ResponseKeys {
        const val photos = "photos"
        const val photo = "photo"
        const val title = "title"
        const val imageSize = "url_m"
    }
}
